### PR TITLE
=doc #1290 Remove custom anchors at the start of the page

### PR DESCRIPTION
**Pull request by jonas**
Paradox does not verify them which makes them hard to maintain and removing them will make it easier to merge the Java and Scala documentation by avoiding to need to fix a lot of additional anchors.

Refs: #1290

### CODE REVIEW

File: docs/src/main/paradox/java/http/routing-dsl/directives/alphabetically.md [hunk 1]
```diff
@@ -1,154 +1,153 @@
...
+|---------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|@ref[authenticateBasic](security-directives/authenticateBasic.md)                                                    | Wraps the inner route with Http Basic authentication support using a given `Authenticator<T>`                                                                                                                       |
+|@ref[authenticateBasicAsync](security-directives/authenticateBasicAsync.md)                                          | Wraps the inner route with Http Basic authentication support using a given `AsyncAuthenticator<T>`                                                                                                                  |
```
Comment (jlprat) [inline docs/src/main/paradox/java/http/routing-dsl/directives/alphabetically.md]: Remove trailing spaces

File: docs/src/main/paradox/java/http/routing-dsl/directives/marshalling-directives/index.md [hunk 2]
```diff
@@ -21,8 +20,8 @@ Marshalling directives usually rely on an in-scope implicit marshaller to handle
...
-|-----------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|
-|@ref[completeWith](completeWith.md#completewith-java) | Uses a marshaller for a given type to produce a completion function for an inner route. Used in conjunction with *instanceOf* to format responses.|
-|@ref[entity](entity.md#entity-java)                   | Unmarshalls the request entity to the given type and passes it to its inner route.  Used in conjunction with *as* to convert requests to objects. |
-|@ref[handleWith](handleWith.md#handlewith-java)       | Completes a request with a given function, using an in-scope unmarshaller for an input and in-scope marshaller for the output.                   |
\ No newline at end of file
+|directive                           | behavior                                                                                                                                          |
```
Comment (jlprat) [inline docs/src/main/paradox/java/http/routing-dsl/directives/marshalling-directives/index.md]: Remove trailing space

File: docs/src/main/paradox/scala/http/routing-dsl/directives/marshalling-directives/index.md [hunk 3]
```diff
@@ -21,8 +20,8 @@ Marshalling directives usually rely on an in-scope implicit marshaller to handle
...
-|-------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|
-|@ref[completeWith](completeWith.md#completewith) | Uses a marshaller for a given type to produce a completion function for an inner route. Used in conjunction with *instanceOf* to format responses.|
-|@ref[entity](entity.md#entity)                   | Unmarshalls the request entity to the given type and passes it to its inner route.  Used in conjunction with *as* to convert requests to objects. |
-|@ref[handleWith](handleWith.md#handlewith)       | Completes a request with a given function, using an in-scope unmarshaller for an input and in-scope marshaller for the output.                   |
\ No newline at end of file
+|directive                           | behavior                                                                                                                                          |
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/marshalling-directives/index.md]: I think there is a trailing space here

**Comment by raboof** [comment]
Cool! I see you also removed them from the `@ref`s, did you make sure not to remove references to headings in the middle of the target page?

**Comment by jonas** [comment]
@raboof I used this script to do most of the work. It only considers anchors present at the start of any page.

```sh
anchors=$(find docs/src/main/paradox/ -name "*.md" | xargs -n 1 head -1 | sed -n 's#<a id="\(.*\)"></a>#\1#p')
files=$(find docs/src/main/paradox/ -name "*.md")
# Remove fragments from links
for id in $anchors; do sed -i "s/#$id[)]/)/" $files; done
# Remove top-level page anchors
for id in $anchors; do sed -i "/^<a.id=.$id.><.a>$/d" $files; done
```

**Comment by jlprat** [comment]
Regarding my trailing spaces comments, I think it was just github diff UI that fooled me.

**Comment by raboof** [comment]
@jonas ah didn't notice it was in the commit message, too - neat!

File: docs/src/main/paradox/java/http/routing-dsl/directives/alphabetically.md (hunk 1 above)
Comment (jonas) [inline docs/src/main/paradox/java/http/routing-dsl/directives/alphabetically.md]: I was not able to find any trailing spaces here nor for the two other cases you pointed out. How can I reveal them? `git show` doesn't display a red box at the end of the line as it normally does for trailing spaces.

**Comment by jonas** [comment]
@jlprat 👍 . There are tons of trailing spaces in the documentation though and it might be good to remove them at some point.

**Comment by jonas** [comment]
@jrudolph The page-level anchors were never useful, but there because we had to convert all RST references and anchors for correct cross-referencing. I was considering adding some checks to paradox but that never happened.

**Comment by raboof** [comment]
Checking references (https://github.com/lightbend/paradox/issues/120) and removing trailing whitespace scalariform-style (https://github.com/lightbend/paradox/issues/103) would indeed be nice!

**Comment by jrudolph** [comment]
@jonas cool, thanks for the clarification.

**Pull request merged**